enhancements/cguConditionEnhancement.md

```diff
@@ -0,0 +1,452 @@
+
+This section highlights the recommendation on condition types from kubernetes community and what other operators are following the design pattern in general.
+
+- Article provided by @ricardo: [Conditions in Kubernetes Controller](https://maelvls.dev/kubernetes-conditions/)
+- [Cert-manager: Automatically provision and manage TLS certificates in Kubernetes](https://github.com/cert-manager/cert-manager) uses the single Ready condition which matches to the current TALM condition type
+- [Kubernetes recommendation on Conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+- Posted a question in slack channel : #forum-apiserver for any existing design guidelines. Still waiting for some feedback.
+- **Openshift-apiserver, kube-apiserver, machine-config, node-tuning, config-operator** have below conditions:
+
+```yaml
+[root@hv2 ~]# oc get co/openshift-apiserver -oyaml | yq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:38:09Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:39:58Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:43:16Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Available"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:38:08Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Upgradeable"
+  }
+]
+```
+
+- **Network operator** has below condition:
+
+```yaml
+[root@hv2 ~]# oc get co/network -ojson | jq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "ManagementStateDegraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "True",
+    "type": "Upgradeable"
+  },
+  {
+    "lastTransitionTime": "2022-06-07T15:35:25Z",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:37:57Z",
+    "status": "True",
+    "type": "Available"
+  }
+]
+```
+
+## Proposal
+
+This section proposes minimalistic changes without breaking the current design pattern. The below proposal/recommendation consists of more granular `status.conditions`  based on all the verification we already have in TALM.
+
+**Considering**: backup,precaching is true and enable for upgrade is true as well. 
+
+### Workflow Description
+
+1. Condition type related to cluster existing in the managedcluster. The type could reflect on the existence of the targeted cluster to be present in the managedcluster. 
+
+  ```yaml
+  Type:    "ClusterValidation/ClustersSelected",
```

Comment:
Maybe, to avoid any confusion, "TargetClustersValidated" of "SelectedClustersValidated"?

Reply:
I would vote for SelectedClustersValidated. Thanks

Reply:
@imiller0 After a general discussion, i opted to use ClustersSelected